src/Sarif.Multitool.Library/MergeCommand.cs

```diff
@@ -235,7 +235,7 @@ private void ProcessInputSarifLog(string filePath)
                 new FixupVisitor().VisitSarifLog(sarifLog);
             }
 
-            _mergeLogsChannel.Writer.WriteAsync(sarifLog);
+            _mergeLogsChannel.Writer.TryWrite(sarifLog);
```

Comment:
Does it matter that we've gone from an asynchronous command to a synchronous one?  Also, we don't do anything with the boolean indicating whether or not the write was successful.

Reply:
using async method without await or task could lead to some issues: "CA2012 ValueTask instances returned from method calls should always be used, typically awaited. Not doing so often represents a functional bug, but even if it doesn't, it can result in degraded performance if the target method pools objects for use with ValueTasks.".

---
In reply to: [562075944](https://github.com/microsoft/sarif-sdk/pull/2253#discussion_r562075944) [](ancestors = 562075944)

Reply:
That makes sense.  Are you concerned that the write might fail and we don't tell anyone about it?

---
In reply to: [562084169](https://github.com/microsoft/sarif-sdk/pull/2253#discussion_r562084169) [](ancestors = 562084169,562075944)

Reply:
for now, no. if that happens, we will have to think carefully, because we would need to create a retry logic. or something related

---
In reply to: [562087447](https://github.com/microsoft/sarif-sdk/pull/2253#discussion_r562087447) [](ancestors = 562087447,562084169,562075944)